internal/testing/backend/search_test.go

```diff
@@ -0,0 +1,143 @@
+//go:build integration
+
+package backend_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/guacsec/guac/internal/testing/testdata"
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+)
+
+func TestFindSoftware(t *testing.T) {
+	b := setupTest(t)
+	tests := []struct {
+		name       string
+		searchText string
+		want       []model.PackageSourceOrArtifact
+		wantErr    bool
+		InPkg      []*model.PkgInputSpec
+		InSrc      []*model.SourceInputSpec
+		InArt      []*model.ArtifactInputSpec
+	}{
+		{
+			name:       "default package",
+			InPkg:      []*model.PkgInputSpec{testdata.P1},
+			searchText: "tensorflow",
+			want: []model.PackageSourceOrArtifact{
+				model.Package{
+					Type: "pypi",
+					Namespaces: []*model.PackageNamespace{{
+						Names: []*model.PackageName{{
+							Name: "tensorflow",
+							Versions: []*model.PackageVersion{{
+								Version:    "",
+								Qualifiers: []*model.PackageQualifier{},
+							}},
+						}},
+					},
+					},
+				}},
+		},
+		{
+			name:       "package no match",
+			InPkg:      []*model.PkgInputSpec{testdata.P1},
+			searchText: "invalid",
+			want:       []model.PackageSourceOrArtifact{},
+		},
+		{
+			name: "default artifact",
+			InArt: []*model.ArtifactInputSpec{
+				{
+					Algorithm: "sha256",
+					Digest:    "testdigest",
+				},
+			},
+			searchText: "test",
+			want: []model.PackageSourceOrArtifact{
+				&model.Artifact{
+					Algorithm: "sha256",
+					Digest:    "testdigest",
+				},
+			},
+		},
+		{
+			name: "artifact no match",
+			InArt: []*model.ArtifactInputSpec{
+				{
+					Algorithm: "sha256",
+					Digest:    "testdigest",
+				},
+			},
+			searchText: "invalid",
+			want:       []model.PackageSourceOrArtifact{},
+		},
+		{
```

Comment:
can you also add a test where the `package` and `source` are both returned based on the `searchText`